Chummer/Classes/clsCustomDataDirectoryInfo.cs

```diff
@@ -0,0 +1,475 @@
+                    string.Format(GlobalOptions.CultureInfo,
+                        LanguageManager.GetString("Message_FailedLoad"), ex.Message),
+                    string.Format(GlobalOptions.CultureInfo,
+                        LanguageManager.GetString("MessageTitle_FailedLoad")
+                        + " " + Name),
+                    MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+            finally
+            {
+                _blnCreated = true;
+            }
+        }
+        #endregion
+
+        /// <summary>
+        /// Checks if any custom data is activated, that matches name and version of the dependent upon directory
+        /// </summary>
+        /// <param name="objCharacterOptions"></param>
+        /// <returns>List of the names of all missing dependencies as a single string</returns>
+        public string CheckDependency(CharacterOptions objCharacterOptions)
+        {
+            var missingDependencyList = new List<string>();
+            foreach (var (depGuid, depName, depVersion, isMinimumVersion) in DependenciesList)
+            {
+                if (isMinimumVersion)
+                {
+                    if (objCharacterOptions.EnabledCustomDataDirectoryInfos.All(loadedFile =>
+                        depGuid != loadedFile.Guid && depVersion != loadedFile.Version))
+                        missingDependencyList.Add(string.Format(depName + LanguageManager.GetString("AddMinimumVersion"), depVersion));
+                }
+                else
+                {
+                    if (objCharacterOptions.EnabledCustomDataDirectoryInfos.All(loadedFile =>
+                        depGuid != loadedFile.Guid && depVersion >= loadedFile.Version))
+                        missingDependencyList.Add(string.Format(depName + LanguageManager.GetString("AddVersion"), depVersion));
+                }
+
+            }
+            var missingDependencies = string.Join(", ", missingDependencyList);
+            return string.IsNullOrEmpty(missingDependencies) ? string.Empty : missingDependencies;
+        }
+
+        /// <summary>
+        /// Checks if any custom data is activated, that matches name and version of the prohibited directories
+        /// </summary>
+        /// <param name="objCharacterOptions"></param>
+        /// <returns>List of the names of all prohibited custom data directories as a single string</returns>
+        public string CheckExclusivity(CharacterOptions objCharacterOptions)
+        {
+
+            var existingProhibitedDirectory = new List<string>();
+
+            foreach (var (forbiddenGuid, forbiddenName, forbiddenVersion, ignoreVersion) in ExclusivitiesList)
+            {
+                if (!ignoreVersion)
+                {
+                    if(objCharacterOptions.EnabledCustomDataDirectoryInfos.Any(enabledDirectory =>
+                        enabledDirectory.Guid == forbiddenGuid && enabledDirectory.Version == forbiddenVersion))
```

Comment:
I kind of feel like equality for the version here wouldn't work practically. If version is important for exclusivity information, I would imagine there could be cases where you'd want to forbid any version older than a specific one, any version newer than a specific one, or both. Maybe some sort of wildcard handling might be better, and also work better with dependency checks, where you also could have similar setups (wanting to be dependent on a specific range of versions.

Reply:
There are two versions of this that I'm thinking about right now.

1. First would be to implement some handling for operators.
What I mean by that is, that one could enter `<version>{biggerthen}</version>`, `<version>{equal}</version>`, `<version>{smallerthen}</version>`. I'm not sure how this can be done, but for that I can take a look at how it's already done at other places.

2. The other one would be just having both a <minimumversion> and <maximumversion> node.

Reply:
If you want to implement it in a sane way (and are willing to put some work into it), the guys from nuget had to come up with a robust way to handle all this version dependencies (probably in a much more clever way than we could in years).

So - if you are "inventing" a syntax I would suggest to take a look at their "Version ranges" syntax for some ideas.

https://docs.microsoft.com/en-us/nuget/concepts/package-versioning

Reply:
That's a nice syntax for sure, if I find a nice way to implement and handle it I will just use that.

But maybe just using inclusive min and max versions without any syntax is enough and way easier to use.